CI fails with dev dependencies: RuntimeError: Could not infer dtype of dict

CI fails with dev dependencies: https://github.com/huggingface/trl/actions/runs/19049362324/job/54405508558
> RuntimeError: Could not infer dtype of dict
```python
  FAILED tests/test_reward_trainer.py::TestRewardTrainer::test_train_with_set_chat_template_from_model - RuntimeError: Could not infer dtype of dict
  FAILED tests/test_reward_trainer.py::TestRewardTrainer::test_train_dataset_types[conversational_preference] - RuntimeError: Could not infer dtype of dict
  FAILED tests/test_reward_trainer.py::TestRewardTrainer::test_train_with_set_chat_template_from_path - RuntimeError: Could not infer dtype of dict
  FAILED tests/test_reward_trainer.py::TestRewardTrainer::test_train_dataset_types[conversational_implicit_prompt_preference] - RuntimeError: Could not infer dtype of dict
  = 4 failed
```
Stacktrace:
```python
   >       trainer.train()
  
  tests/test_reward_trainer.py:165: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  .venv/lib/python3.12/site-packages/transformers/trainer.py:2155: in train
      return inner_training_loop(
  .venv/lib/python3.12/site-packages/transformers/trainer.py:2430: in _inner_training_loop
      batch_samples, num_items_in_batch = self.get_batch_samples(epoch_iterator, num_batches, args.device)
                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/transformers/trainer.py:5154: in get_batch_samples
      batch_samples.append(next(epoch_iterator))
                           ^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/accelerate/data_loader.py:567: in __iter__
      current_batch = next(dataloader_iter)
                      ^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/torch/utils/data/dataloader.py:732: in __next__
      data = self._next_data()
             ^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/torch/utils/data/dataloader.py:788: in _next_data
      data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/torch/utils/data/_utils/fetch.py:55: in fetch
      return self.collate_fn(data)
             ^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/transformers/data/data_collator.py:42: in __call__
      return self.torch_call(features)
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  self = DataCollatorForPreference(pad_token_id=151643, pad_to_multiple_of=None, return_tensors='pt')
  examples = [{'chosen_input_ids': {'attention_mask': [1, 1, 1, 1, 1, 1, ...], 'input_ids': [151644, 8948, 198, 2610, 525, 1207, ....._input_ids': {'attention_mask': [1, 1, 1, 1, 1, 1, ...], 'input_ids': [151644, 8948, 198, 2610, 525, 1207, ...]}}, ...]
  
      def torch_call(self, examples: list[dict[str, Any]]) -> dict[str, Any]:
          # Convert to tensor
  >       chosen_input_ids = [torch.tensor(example["chosen_input_ids"]) for example in examples]
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  E       RuntimeError: Could not infer dtype of dict
  
  trl/trainer/reward_trainer.py:144: RuntimeError
```

--- a/trl/trainer/reward_trainer.py
+++ b/trl/trainer/reward_trainer.py
@@ -488,13 +488,15 @@ class RewardTrainer(BaseTrainer):
                         chosen_input_ids = processing_class.apply_chat_template(
                             example["chosen"],
                             tools=example.get("tools"),
+                            return_dict=True,
                             **example.get("chat_template_kwargs", {}),
-                        )
+                        )["input_ids"]
                         rejected_input_ids = processing_class.apply_chat_template(
                             example["rejected"],
                             tools=example.get("tools"),
+                            return_dict=True,
                             **example.get("chat_template_kwargs", {}),
-                        )
+                        )["input_ids"]
                         output = {"chosen_input_ids": chosen_input_ids, "rejected_input_ids": rejected_input_ids}
                     else:
                         output = {

